### partG-Network initialization


In [1]:
from __future__ import division, print_function, absolute_import

import tflearn
from tflearn.layers.core import input_data, dropout, fully_connected,flatten
from tflearn.layers.conv import conv_1d, max_pool_1d,conv_2d, max_pool_2d
from tflearn.layers.normalization import local_response_normalization
from tflearn.layers.estimator import regression

import tensorflow as tf
import numpy as np

In [2]:
data = np.genfromtxt("norm.csv",delimiter=",",missing_values="?",filling_values=1.)
#Separate the X
dataX = data[:,1:-1]
#Separate the Y
##first column record whether the team win the game
pre_dataY = data[:,0] 
#Convert the Y to one hot
dataY = np.zeros((pre_dataY.size,2))


for i in range(len(pre_dataY)):
    if pre_dataY[i] >= 15:
        dataY[i][0] = 1
    else:
        dataY[i][1] = 1


In [3]:
dataX = dataX.reshape([-1, 2, 3, 1])
dataX.shape


(87898, 2, 3, 1)

In [4]:
dataY.shape

(87898, 2)

In [5]:
#Number of data points used for testing
num_test = int(0.15 * len(data))

#Split data into train and  test
trainX = dataX[:-num_test]
testX = dataX[-num_test:]

trainY = dataY[:-num_test]
testY = dataY[-num_test:]


In [6]:
network5 = input_data(shape=[None, 2, 3, 1], name='input')
network5 = conv_2d(network5,4,2,activation='relu',regularizer="L2")
network5 = max_pool_2d(network5,1)
network5 = fully_connected(network5, 16, activation='tanh')
#network5 = dropout(network1,0.8)
network5 = fully_connected(network5, 2, activation='tanh')
network5 = regression(network5, optimizer='adam', learning_rate=0.01,loss='mean_square', name='target')


In [7]:
tflearn.initializations.uniform (shape=None, minval=0, maxval=None, dtype=tf.float32, seed=None)

In [8]:
model = tflearn.DNN(network5, tensorboard_verbose=3)
model.fit({'input': trainX}, {'target': trainY}, n_epoch=20,
           validation_set=({'input': testX}, {'target': testY}),
           snapshot_step=100, show_metric=True, run_id='Ruohua Yin')


Training Step: 23359  | total loss: 0.01372 | time: 17.305s
| Adam | epoch: 020 | loss: 0.01372 - acc: 0.9822 -- iter: 74688/74714
Training Step: 23360  | total loss: 0.01239 | time: 18.379s
| Adam | epoch: 020 | loss: 0.01239 - acc: 0.9840 | val_loss: 0.00486 - val_acc: 0.9954 -- iter: 74714/74714
--


<img src="conv2d.png">

### Change the network architecture. How does it effect the accuracy?

** the accuracy decline from 0.9941 to 0.9840

### How does it effect how quickly the network plateaus?

** the plateaus become longer.